nvflare/apis/client_engine_spec.py

```diff
@@ -64,40 +64,3 @@ def get_all_components(self) -> dict:
 
     def get_widget(self, widget_id: str) -> Widget:
```

Comment:
also suggest add `@abstractmethod` to all the methods that we require the child class to implement